tests/smoke/APNSClientSmokeTest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+declare( strict_types = 1 );
+
+/**
+ * A smoke test to find basic compilation errors in `APNSClient.php`
+ *
+ * @covers APNSClient
+ */
+class APNSClientSmokeTest extends APNSTest {
+
+	public function testThatSetPortNumberWorks() {
+		$client = ( new APNSClient( $this->new_configuration() ) )->setPortNumber( random_int( 1, 65535 ) );
+		$this->assertNotNull( $client );
+	}
+
+	public function testThatSetDebugWorks() {
+		$client = ( new APNSClient( $this->new_configuration() ) )->setDebug( true );
+		$this->assertNotNull( $client );
+	}
+
+	public function testThatSetDisableSSLWorks() {
+		$client = ( new APNSClient( $this->new_configuration() ) )->setDisableSSLVerification( true );
+		$this->assertNotNull( $client );
+	}
+
+	public function testThatSetCertificateBundlePathWorks() {
+		$client = ( new APNSClient( $this->new_configuration() ) )->setCertificateBundlePath( $this->random_string() );
+		$this->assertNotNull( $client );
+	}
+
+	public function testThatCloseWorks() {
+		$client = ( new APNSClient( $this->new_configuration() ) )->close();
+		$this->assertNotNull( $client );
+	}
+}
```

Comment:
These are interesting.

I made these changes:

```diff
@@ -53,10 +53,10 @@ class APNSClient {
 			$this->enqueueRequest( $request );
 		}
 
-		return $this->sendQueuedRequests();
+		return $this->sendQueuedRequests_();
 	}
 
-	public function close(): void {
+	public function close_(): void {
 		curl_multi_close( $this->curl_handle );
 	}
```

and I got this output from Psalm:

```
ERROR: UndefinedMethod - examples/run-queue.php:35:10 - Method APNSClient::close does not exist (see https://psalm.dev/022)
$client->close();


ERROR: UndefinedMethod - index.php:59:10 - Method APNSClient::close does not exist (see https://psalm.dev/022)
$client->close();


ERROR: UndefinedMethod - src/APNSClient.php:56:17 - Method APNSClient::sendQueuedRequests_ does not exist (see https://psalm.dev/022)
		return $this->sendQueuedRequests_();


------------------------------
3 errors found
```

So, as you say, Psalm sort of does this check already. **But**... Psalm can only notice if a method of `APNSClient` has been renamed without updating its callers as long as there are callers. In fact, we can see that the errors regarding the `close` method rename occur only in files outside of the source or the tests.

I think there's value in keeping these for the moment, I've done my fair share of fat-fingered mistakes in my time.

It'd be interesting to see how much of these we can translate in appropriate tests as I (very slowly) progress with the [integration testing experiment](https://github.com/Automattic/php-push/tree/add/integration-tests).